sql/parse/dateparse/date.go

```diff
@@ -6,6 +6,14 @@ import (
 	"time"
 )
 
+type OutType int
```

Comment:
`OutType` feels a little bit too generic of a name. 

Actually... you might see how it looks to reuse the existing `Date`, `Time`, and `DateTime` type classes and see if you can just pass those around and avoid having to create a new type. If we can sensibly reuse something, that'd be my first choice.